File: 2487-remove-nodes-from-linked-list/2487-remove-nodes-from-linked-list.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode removeNodes(ListNode head) {
        ListNode temp = head;
        Stack<Integer> stack = new Stack<>();
        while (temp != null) {
            while (!stack.isEmpty() && temp.val > stack.peek()) {
                stack.pop();
            }
            stack.push(temp.val);
            temp = temp.next;
        }
        ListNode dumm = new ListNode();
        temp = dumm;
        for (int i : stack) {
            temp.next = new ListNode(i);
            temp = temp.next;
        }
        return dumm.next;
        
    }
}

File: 2487-remove-nodes-from-linked-list/2487-remove-nodes-from-linked-list.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def removeNodes(self, head: Optional[ListNode]) -> Optional[ListNode]:
        temp = head
        stack = []
        while temp:
            while len(stack)!=0 and temp.val>stack[-1]:
                stack.pop()
            stack.append(temp.val)
            temp = temp.next
        dumm = ListNode()
        temp = dumm
        for i in stack:
            temp.next = ListNode(i)
            temp = temp.next
        return dumm.next;
        